src/browser/modules/Stream/CypherFrame/helpers.js

```diff
@@ -287,3 +298,110 @@ const arrayifyPath = (types = neo4j.types, path) => {
     ].filter(part => part !== null)
   })
 }
+
+/**
+ * Converts a raw Neo4j record into a JSON friendly format, mimicking APOC output
+ * @param     {Record}    record
+ * @return    {*}
+ */
+export function recordToJSONMapper (record) {
+  const keys = get(record, 'keys', [])
+
+  return reduce(
+    keys,
+    (agg, key) => {
+      const field = record.get(key)
+
+      return {
+        ...agg,
+        [key]: mapNeo4jValuesToPlainValues(field)
+      }
+    },
+    {}
+  )
+}
+
+/**
+ * Recursively converts Neo4j values to plain values, leaving other types untouched
+ * @param     {*}     values
+ * @return    {*}
+ */
+function mapNeo4jValuesToPlainValues (values) {
+  if (!isObjectLike(values)) {
+    return values
+  }
+
+  if (Array.isArray(values)) {
+    return map(values, mapNeo4jValuesToPlainValues)
+  }
+
+  if (isNeo4jValue(values)) {
+    return neo4jValueToPlainValue(values)
+  }
+
+  // could be a Node or Relationship
+  const type = lowerCase(get(values, 'constructor.name', ''))
+
+  if (includes(['relationship', 'node'], type)) {
+    const labels =
```

Comment:
From discussion: Call this `type` from relationships and keep `labels` for nodes.